Tools/YamlCreate.ps1

```diff
@@ -31,232 +31,663 @@ filter TrimString {
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Privacy Url'
+            $PrivacyUrl = Read-Host -Prompt 'Privacy Url' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PrivacyUrl) -and ($PrivacyUrl.Length -lt 5 -or $PrivacyUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Application Author'
+            $Author = Read-Host -Prompt 'Author' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($Author) -and ($Author.Length -lt 1 -or $Author.Length -gt 256))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Url to the homepage of the application.'
+            $PackageUrl = Read-Host -Prompt 'Homepage' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PackageUrl) -and ($PackageUrl.Length -lt 5 -or $PackageUrl.Length -gt 2000))
+
+        while ([string]::IsNullOrWhiteSpace($License) -or $License.Length -ge 512) {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the License. For example: MIT, GPL, Freeware, Proprietary or Copyright (c) Microsoft Corporation'
+            $License = Read-Host -Prompt 'License' | TrimString
+        }
+        
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the License URL.'
+            $LicenseUrl = Read-Host -Prompt 'License URL' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($LicenseUrl) -and ($LicenseUrl.Length -lt 10 -or $LicenseUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Copyright'
+            $Copyright = Read-Host -Prompt 'Copyright' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($Copyright) -and ($Copyright.Length -lt 5 -or $Copyright.Length -gt 512))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Copyright Url'
+            $CopyrightUrl = Read-Host -Prompt 'CopyrightUrl' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($CopyrightUrl) -and ($LicenseUrl.Length -lt 10 -or $LicenseUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter any tags that would be useful to discover this tool. For example: zip, c++ (Max 16)'
+            $Tags = Read-Host -Prompt 'Tags' | TrimString
+        } while ($Tags.Split(", ").Count -gt '16')
+
+        while ([string]::IsNullOrWhiteSpace($ShortDescription) -or $ShortDescription.Length -gt '256') {
+            Write-Host ''
```

Comment:
can be simplified to just `Write-Host`

Reply:
I see.. Not sure why i always do '' or "" after write-host for a new line, the Original YamlCreate used 

`$NewLine = [System.Environment]::NewLine`
For a double space..